### PR TITLE
feat: TabGroup component

### PR DESCRIPTION
**Pull request by RohitRaj011**
# Description

Tab Group Component Creation

[Design](https://www.figma.com/design/85LCN9lsyBoXjpdczYP1Ti/Devtron-Component-Library?node-id=4340-2928&t=6hgQHAILf0pjrT0w-0)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [x] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Shared/Components/TabGroup/TabGroup.types.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    tabs: TabProps[]
+    /**
+     * Size of the tabs.
+     * @default ComponentSizeType.large
+     */
+    size?: ComponentSizeType.large | ComponentSizeType.medium
+    /**
+     * Optional component to be rendered on the right side of the tab list.
+     */
+    rightComponent?: React.ReactNode
```
Comment (eshankvaish) [inline src/Shared/Components/TabGroup/TabGroup.types.ts:107]:
```suggestion
    rightComponent?: ReactElement
```

File: src/Shared/Components/TabGroup/TabGroup.types.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import { DataAttributes } from '@Shared/types'
+
+type TabComponentProps<TabTypeProps> = TabTypeProps & DataAttributes
+
+type ConditionalTabType =
+    | {
+          /**
+           *  Type of the tab, either `button`, `link` or `navLink`.
+           */
+          tabType: 'button'
```
Comment (eshankvaish) [inline src/Shared/Components/TabGroup/TabGroup.types.ts:13]: suggestion: can add enum

File: src/Shared/Components/TabGroup/TabGroup.scss [hunk 3]
```diff
@@ -0,0 +1,107 @@
+.tab-group {
+    list-style: none;
+
+    &__tab {
+        position: relative;
+
+        &::after {
```
Comment (eshankvaish) [inline src/Shared/Components/TabGroup/TabGroup.scss:7]:
```suggestion
        $parent-selector: &;
```

File: src/Shared/Components/TabGroup/TabGroup.scss [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        }
+
+        &:hover {
+            color: var(--B500);
+
+            .tab-group__tab__icon {
+                &--stroke {
+                    stroke: var(--B500);
+
+                    path {
```
Comment (eshankvaish) [inline src/Shared/Components/TabGroup/TabGroup.scss:30]:
```suggestion
            #{$parent-selector}__icon {
```

File: src/Shared/Components/TabGroup/TabGroup.scss [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                    fill: var(--B500);
+
+                    path {
+                        fill: var(--B500);
+                    }
+                }
+            }
+        }
+
+        &--active::after {
```
Comment (eshankvaish) [inline src/Shared/Components/TabGroup/TabGroup.scss:45]: let's not add any of fill and stroke when icon type is null for non-path elements

File: src/Shared/Components/TabGroup/TabGroup.scss [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+        &__warning-icon {
+            > g path:nth-of-type(2) {
+                fill: var(--Y700);
+            }
+        }
+
+        &__nav-link {
+            &.active {
+                color: var(--B500);
```
Comment (eshankvaish) [inline src/Shared/Components/TabGroup/TabGroup.scss:72]: warning-icon-y7

**Review by eshankvaish — CHANGES_REQUESTED** [review]

**Review by eshankvaish — APPROVED** [review]

**Review by eshankvaish — DISMISSED** [review]

File: src/Shared/Components/CICDHistory/LogsRenderer.tsx [hunk 7]
```diff
@@ -408,7 +408,7 @@ export const LogsRenderer = ({
                     }}
                 >
                     <div
-                        className={`flexbox-col pb-7 dc__position-sticky dc__zi-2 ${fullScreenView ? 'dc__top-0' : 'dc__top-38'}`}
+                        className={`flexbox-col pb-7 dc__position-sticky dc__zi-2 ${fullScreenView ? 'dc__top-0' : 'dc__top-36'}`}
```
Comment (eshankvaish) [inline src/Shared/Components/CICDHistory/LogsRenderer.tsx:411]: FYI, i have also added top 36 in dashboard

File: src/Shared/Components/TabGroup/TabGroup.component.tsx [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    size,
+    badge = null,
+    alignActiveBorderWithContainer,
+    hideTopPadding,
+    showIndicator,
+    showError,
+    showWarning,
+    disabled,
+}: TabProps & Pick<TabGroupProps, 'size' | 'alignActiveBorderWithContainer' | 'hideTopPadding'>) => {
+    const getClassNameBySize = () => {
```
Comment (vivek-devtron) [inline src/Shared/Components/TabGroup/TabGroup.component.tsx:26]: Please extract out the utility methods

File: src/Shared/Components/TabGroup/TabGroup.component.tsx [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+        </li>
+    )
+}
+
+export const TabGroup = ({
+    tabs = [],
+    size = ComponentSizeType.large,
+    rightComponent,
+    alignActiveBorderWithContainer,
+    hideTopPadding,
```
Comment (vivek-devtron) [inline src/Shared/Components/TabGroup/TabGroup.component.tsx:118]: We should add the support for dataTestId

File: src/Shared/Components/CICDHistory/LogsRenderer.tsx (hunk 7 above)
Comment (RohitRaj011) [reply src/Shared/Components/CICDHistory/LogsRenderer.tsx:411]: using from dashboard only

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**